tools/src/test/java/org/apache/kafka/tools/GroupsCommandTest.java

```diff
@@ -357,6 +382,122 @@ public void testListGroupsFailsWithException() {
+                        List.of("--bootstrap-server", clusterInstance.bootstrapServers(), "--list", "--group-type", "classic").toArray(new String[0])))));
+                if (res.getKey().split("\n").length == 3 && res.getValue().isEmpty()) {
+                    assertCapturedListOutput(res.getKey(),
+                        new String[]{classicGroupId, "Classic", "consumer"},
+                        new String[]{simpleGroupId, "Classic"});
+                    return true;
+                }
+                return false;
+            }, "Waiting for listing groups to return classic type groups");
+
+            TestUtils.waitForCondition(() -> {
+                Map.Entry<String, String> res = ToolsTestUtils.grabConsoleOutputAndError(() ->
+                    assertDoesNotThrow(() -> groupsCommand.listGroups(new GroupsCommand.GroupsCommandOptions(
+                        List.of("--bootstrap-server", clusterInstance.bootstrapServers(), "--list", "--group-type", "consumer").toArray(new String[0])))));
+                if (res.getKey().split("\n").length == 2 && res.getValue().isEmpty()) {
+                    assertCapturedListOutput(res.getKey(),
+                        new String[]{consumerGroupId, "Consumer", "consumer"});
+                    return true;
+                }
+                return false;
+            }, "Waiting for listing groups to return consumer type groups");
+
+            TestUtils.waitForCondition(() -> {
+                Map.Entry<String, String> res = ToolsTestUtils.grabConsoleOutputAndError(() ->
+                    assertDoesNotThrow(() -> groupsCommand.listGroups(new GroupsCommand.GroupsCommandOptions(
+                        List.of("--bootstrap-server", clusterInstance.bootstrapServers(), "--list", "--group-type", "share").toArray(new String[0])))));
+                if (res.getKey().split("\n").length == 2 && res.getValue().isEmpty()) {
+                    assertCapturedListOutput(res.getKey(),
+                        new String[]{shareGroupId, "Share", "share"});
+                    return true;
+                }
+                return false;
+            }, "Waiting for listing groups to return share type groups");
+
+            TestUtils.waitForCondition(() -> {
+                Map.Entry<String, String> res = ToolsTestUtils.grabConsoleOutputAndError(() ->
+                    assertDoesNotThrow(() -> groupsCommand.listGroups(new GroupsCommand.GroupsCommandOptions(
+                        List.of("--bootstrap-server", clusterInstance.bootstrapServers(), "--list", "--share").toArray(new String[0])))));
+                if (res.getKey().split("\n").length == 2 && res.getValue().isEmpty()) {
+                    assertCapturedListOutput(res.getKey(),
+                        new String[]{shareGroupId, "Share", "share"});
+                    return true;
+                }
+                return false;
+            }, "Waiting for listing groups to return share groups");
```

Comment:
nit: To match the other cases, `"Waiting for listing groups to return share type groups"`.

Reply:
Updated it. Thanks for the review.